src/test/timedata_tests.cpp

```diff
@@ -96,9 +96,10 @@ BOOST_AUTO_TEST_CASE(addtimedata)
     // not to fix this because it prevents possible attacks. See the comment in AddTimeData() or issue #4521
     // for a more detailed explanation.
     MultiAddTimeData(2, 100); // filter median is 100 now, but nTimeOffset will not change
+    // We want this test to end with nTimeOffset==0, otherwise subsequent tests of the suite will fail.
     BOOST_CHECK_EQUAL(GetTimeOffset(), 0);
 
-    // We want this test to end with nTimeOffset==0, otherwise subsequent tests of the suite will fail.
+    ResetTimeData();
```

Comment:
in 9d8bb13b1dfc5103e2678ba803ecd3220dce052d: Shouldn't the reset be before the check equal?

Reply:
I don't think so. The `BOOST_CHECK_EQUAL(GetTimeOffset(), 0)` check is to ensure that the net result from all the previous calls is 0. This is still true and better keep that check to ensure it remains so. Surely the check will also pass after `ResetTimeData()` not because the net result is 0, but because the reset obliterated the entire state.